subsys/net/lib/sockets/sockets.c

```diff
@@ -184,6 +226,17 @@ static inline int z_vrfy_zsock_socket(int family, int type, int proto)
 #include <syscalls/zsock_socket_mrsh.c>
 #endif /* CONFIG_USERSPACE */
 
+#ifdef CONFIG_USERSPACE
+static inline int z_vrfy_zsock_socketpair(int family, int type, int proto, int sv[2])
+{
+	/* implementation call to net_context_get() should do all necessary
+	 * checking
+	 */
```

Comment:
You have to check if the calling thread has permission to `sv`.  So, here you have to create a temporary copy, call the implementation with this copy and after it return you copy the memory back using `z_user_to_copy`

Reply:
```
int ret;
int tmp[2];

ret = z_impl_zsock_socketpair(family, type, proto, tmp);
Z_OOPS(z_user_to_copy(sv, tmp, sizeof(tmp));
return ret;
```

Reply:
Thanks @ceolin !